platform-hub-web/src/app/shared/hub-api/hub-api.service.js

```diff
@@ -721,32 +730,73 @@ export const hubApiService = function ($rootScope, $http, $q, logger, events, ap
       });
   }
 
-  function deleteKubernetesToken(userId, cluster) {
-    if (_.isNull(userId) || _.isEmpty(userId)) {
-      throw new Error('"userId" argument not specified or empty');
+  function deleteKubernetesToken(tokenId) {
+    if (_.isNull(tokenId) || _.isEmpty(tokenId)) {
+      throw new Error('"tokenId" argument not specified or empty');
     }
 
-    if (_.isNull(cluster) || _.isEmpty(cluster)) {
-      throw new Error('"cluster" argument not specified or empty');
+    return $http
+      .delete(`${apiEndpoint}/kubernetes/tokens/${tokenId}`)
+      .then(response => {
+        // handle 4xx errors which are not picked up by `catch`.
+        if (response.data.error && response.data.error.status.toString().match(/4../)) {
```

Comment:
Just realised that this should be `if (response.status.toString().match(/4../))`, in case of 4xx errors that don't contain a body.